core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -951,6 +988,31 @@ public ServerBuilder gracefulShutdown(GracefulShutdown gracefulShutdown) {
         return this;
     }
 
+    /**
+     * Sets the {@link GracefulShutdown} configuration for the boss {@link EventLoopGroup}s.
+     * If not set, {@link GracefulShutdown#disabled()} is used.
+     */
+    @UnstableApi
+    public ServerBuilder bossGroupGracefulShutdown(GracefulShutdown gracefulShutdown) {
+        bossGroupGracefulShutdown = requireNonNull(gracefulShutdown, "gracefulShutdown");
+        return this;
+    }
+
+    /**
+     * Sets the function that returns an {@link Throwable} to terminate a pending request when
+     * the server is shutting down. The exception will be converted to an {@link HttpResponse}
+     * by {@link ServerErrorHandler}. If null is returned, the request will be terminated with
+     * {@link ShuttingDownException} that will be converted to an {@link HttpStatus#SERVICE_UNAVAILABLE}
+     * response.
+     */
+    @UnstableApi
+    public ServerBuilder gracefulShutdownExceptionFactory(
```

Comment:
I think specifying `gracefulShutdownExceptionFactory` only makes sense when `ServerBuilder#gracefulShutdown` is specified.

What do you think of adding a single overloaded method which also accepts `gracefulShutdownExceptionFactory`?

e.g.
```java
    /**
     * Sets the {@link GracefulShutdown} configuration.
     * If not set, {@link GracefulShutdown#disabled()} is used.
     */
    @UnstableApi
    public ServerBuilder gracefulShutdown(GracefulShutdown gracefulShutdown, GracefulShutdownExceptionFactory toException) {
        this.gracefulShutdown = requireNonNull(gracefulShutdown, "gracefulShutdown");
        this.toException = requireNonNull(toException, "toException");
        return this;
    }
```